packages/core/src/preflight-notifications.ts

```diff
@@ -0,0 +1,242 @@
+import BigNumber from 'bignumber.js'
+import { nanoid } from 'nanoid'
+import defaultCopy from './i18n/en.json'
+import type { Network } from 'bnc-sdk'
+
+import type { Notification, PreflightNotificationOptions } from './types'
+import { addNotification, removeNotification } from './store/actions'
+import { state } from './store'
+import { eventToType } from './notify'
+import { networkToChainId } from './utils'
+import { configuration } from './configuration'
+import { validatePreflightNotification } from './validation'
+
+let notificationsArr: Notification[]
+state.select('notifications').subscribe(notifications => {
+  notificationsArr = notifications
+})
+
+export async function preflightNotification(
+  options: PreflightNotificationOptions
+): Promise<string> | null {
+  const { apiKey } = configuration
+
+  if (!apiKey) {
+    console.error(
```

Comment:
I don't think an apiKey is required for the preflight notifications as the Blocknative server is not accessed right?

Reply:
yep good point as well for the preflights, updating